core/src/main/java/io/ddf/Factor.java

```diff
@@ -99,14 +113,15 @@ public Factor(String name, T[] data) throws DDFException {
    * @param engineName
    * @throws DDFException
    */
-  public Factor(String name, T[] data, String engineName) throws DDFException {
+  public Factor(String name, T[] data, String engineName, Schema.ColumnType type) throws DDFException {
     super(name, data, engineName);
+    this.mType= type;
   }
 
 
   private Map<String, Integer> mLevelMap;
-  private List<String> mLevels;
-  private Map<String, Integer> mLevelCounts;
+
+  private List<Object> mLevels;
```

Comment:
if the mLevelMap is instantiated as a LinkedHashMap which reserves the order of inserted key and thus we can get the list of levels from it with in the original order, why do we need this mLevels list?